From b3d33b5533ec91fb678028351f8887a6b0a3c881 Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Thu, 26 Jun 2008 15:33:59 +0800
Subject: [PATCH 026/442] ENGR00058763 MX35 P-ATA/ATAPI linux driver

Unit test for P-ATA/ATAPI support on MX35 3DS board.

Signed-off-by: Richard Zhu <r65037@freescale.com>
---
 test/mxc_hdd_test/autorun-ata.sh | 30 ++++++++++++++++++++++++------
 1 file changed, 24 insertions(+), 6 deletions(-)

diff --git a/test/mxc_hdd_test/autorun-ata.sh b/test/mxc_hdd_test/autorun-ata.sh
index 85ec0fe..ef66b77 100755
--- a/test/mxc_hdd_test/autorun-ata.sh
+++ b/test/mxc_hdd_test/autorun-ata.sh
@@ -14,8 +14,15 @@ run_hdd_case()
 	# Generate Test data
 	dd if=/dev/urandom of=/root/hdd_data bs=512 count=10240
 
-	dd if=/root/hdd_data of=/dev/hda bs=512 count=10240
-	dd if=/dev/hda of=/root/hdd_data1 bs=512 count=10240
+	if [ "$(platform)" = "IMX31_3STACK" ]
+	then
+		dd if=/root/hdd_data of=/dev/hda bs=512 count=10240
+		dd if=/dev/hda of=/root/hdd_data1 bs=512 count=10240
+	else
+	
+		dd if=/root/hdd_data of=/dev/sda bs=512 count=10240
+		dd if=/dev/sda of=/root/hdd_data1 bs=512 count=10240
+	fi
 
 	cmp /root/hdd_data1 /root/hdd_data
 
@@ -33,19 +40,30 @@ run_hdd_case()
 if [ "$(platform)" = "IMX27ADS" ]
 then
 	echo "IMX27ADS"
-else
-	echo "IMX31ADS"
+fi
+if [ "$(platform)" = "IMX31_3STACK" ]
+then
+	echo "IMX31_3STACK"
 	modprobe mxc-ide
 	sleep 5
 	modprobe ide_disk
 	sleep 1
+else
+	echo "IMX35_3STACK"
+	modprobe pata_fsl
+	sleep 5
 fi
 
 # devnode test
-check_devnode "/dev/hda"
+if [ "$(platform)" = "IMX31_3STACK" ]
+then
+	check_devnode "/dev/hda"
+else
+	check_devnode "/dev/sda"
+fi
 
 if [ "$STATUS" = 0 ]; then
-	run_hdd_case 
+	run_hdd_case
 fi
 
 print_status
-- 
1.8.0

